src/s2wasm-main.cpp

```diff
@@ -24,40 +24,99 @@ using namespace cashew;
 using namespace wasm;
 
 namespace wasm {
-int debug = 0;
+struct Options {
```

Comment:
this new stuff could go in a header of its own, then it could eventually be reused by other tools here.